### PR TITLE
add external resource to readme

### PR DESCRIPTION
**Pull request by solegalli**
Adds link to open-source repository for online course on hyperparameter optimization. Repo and course feature Optuna among other libraries.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ In addition, integration modules are available for the following libraries, prov
 * [OptGBM: A scikit-learn compatible LightGBM estimator with Optuna](https://github.com/Y-oHr-N/OptGBM)
 * [PyKEEN](https://github.com/pykeen/pykeen)
 * [RL Baselines Zoo](https://github.com/DLR-RM/rl-baselines3-zoo)
+* [Hyperparameter Optimization for Machine Learning](https://github.com/solegalli/hyperparameter-optimization), online course repo
```
Comment (toshihikoyanase) [inline README.md:122]: How about using the same styles as the other entries?
More specifically, I propose two modifications:
- including the whole text to the link
- describe without abbreviation (i.e., replace repo with repository)

The following is just an example. Please feel free to update it.

```suggestion
* [Hyperparameter Optimization for Machine Learning, code repository for online course](https://github.com/solegalli/hyperparameter-optimization)
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for sharing the materials of your online course. I think the notebooks will help Optuna users. I have a small comment, so please take a look.

**Comment by solegalli** [comment]
@toshihikoyanase thank you! glad you found the notebooks useful

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update! The change looks good to me.

**Pull request merged**